### PR TITLE
Updating discussion block OLX

### PR DESCRIPTION
**Pull request by jolyonb**
## [DOC-3902](https://openedx.atlassian.net/browse/DOC-3902)

This overhauls the discussion block OLX documentation, which is out of date and missing information. I've read through the [corresponding source code](https://github.com/edx/edx-platform/blob/master/openedx/core/lib/xblock_builtin/xblock_discussion/xblock_discussion.py) to figure out what's actually going on, and updated the documentation appropriately. I'm pretty sure I've got it right, but somebody who understands the source code better than me may like to confirm.

I apologize that I don't know how to fill in most of the below information.

### Date Needed

This documents features that were implemented years ago.

### Reviewers

Possible roles follow. The PR submitter checks the boxes after each reviewer finishes and gives :+1:. 

- [x] Subject matter expert: jolyonb
- [ ] Subject matter expert: 
- [x] Doc team review (sanity check, copy edit, or dev edit?): @edx/doc
- [ ] Product review:
- [ ] Partner support: 
- [ ] PM review: 

FYI: Tag anyone else who might be interested in this PR here.

### Testing

- [x] Ran ./run_tests.sh without warnings or errors

### HTML Version (optional)

- [ ] Build an RTD draft for your branch and add a link here

### Sandbox (optional)

- [ ] Point to or build a sandbox for the software change and add a link here

### Post-review

- [ ] Add a comment with the description of this change or link this PR to the next release notes task.
- [ ] Squash commits

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @jolyonb! I've created [OSPR-2306](https://openedx.atlassian.net/browse/OSPR-2306) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by jolyonb** [comment]
Bump?

**Comment by nedbat** [comment]
@dianakhuang Can you be the discussion block expert for this?

File: en_us/olx/source/components/discussion-components.rst [hunk 1]
```diff
@@ -15,40 +15,53 @@ course.
 Create the XML File for a Discussion Component
 **********************************************
 
-You create an XML file in the ``discussion`` directory for each inline
-discussion component in your course.
+There are two ways to create an inline discussion component in your course.
+The first is to include a discussion tag inside a vertical, as demonstrated in
```
Comment (grantgoodmanedX) [inline en_us/olx/source/components/discussion-components.rst]: I think we should do more to point users toward the first method. "The first, preferred, way is to include..."

File: en_us/olx/source/components/discussion-components.rst [hunk 2]
```diff
@@ -15,40 +15,53 @@ course.
...
+      display_name="Peer Grading"
+      discussion_category="Essays"
+      discussion_target="Peer Grading"
+      discussion_id="unique_id_here"
+    />
+  </vertical>
 
-An example of how you create a discussion component in a vertical XML file
-follows.
+The second way is to create an XML file in the ``discussion`` directory for each
```
Comment (grantgoodmanedX) [inline en_us/olx/source/components/discussion-components.rst]: "The second way, which is deprecated, is to create..."

File: en_us/olx/source/components/discussion-components.rst [hunk 3]
```diff
@@ -69,26 +82,25 @@ The ``discussion`` element contains no children.
 
    * - Attribute
      - Meaning
+   * - Required. ``discussion_category``
+     - The name of the category for the inline discussion as shown in the main
+       **Discussion** tab of the course. Eg: ``Problem Set 1``
+   * - Required. ``discussion_target``
+     - The name of the subcategory for the inline discussion as shown in the
+       **Discussion** tab of the course. Eg: ``Problem 2``
```
Comment (grantgoodmanedX) [inline en_us/olx/source/components/discussion-components.rst]: Replace "Eg." with "For example:"

File: en_us/olx/source/components/discussion-components.rst [hunk 4]
```diff
@@ -69,26 +82,25 @@ The ``discussion`` element contains no children.
 
    * - Attribute
      - Meaning
+   * - Required. ``discussion_category``
+     - The name of the category for the inline discussion as shown in the main
+       **Discussion** tab of the course. Eg: ``Problem Set 1``
```
Comment (grantgoodmanedX) [inline en_us/olx/source/components/discussion-components.rst]: Replace "Eg." with "For example:"

File: en_us/olx/source/components/discussion-components.rst [hunk 5]
```diff
@@ -69,26 +82,25 @@ The ``discussion`` element contains no children.
...
-       example. Do not use a period (.) in the ID value.
-   * - ``discussion_id``
-     - TBD
-   * - ``discussion_target``
-     - TBD
-
+       courses in the edX deployment. We recommend that you leave this value
+       blank so that a unique identifier is automatically generated.
+   * - Obsolete. ``id``
+     - Now called ``discussion_id``.
```
Comment (grantgoodmanedX) [inline en_us/olx/source/components/discussion-components.rst]: "Use ``discussion_id` instead.

File: en_us/olx/source/components/discussion-components.rst [hunk 6]
```diff
@@ -69,26 +82,25 @@ The ``discussion`` element contains no children.
...
+   * - Obsolete. ``for``
+     - Now called ``discussion_target``.
```
Comment (grantgoodmanedX) [inline en_us/olx/source/components/discussion-components.rst]: "Use ``discussion_target`` instead."

**Review by grantgoodmanedX — CHANGES_REQUESTED** [review]
Thanks for contributing this. I have a few small edits.

**Review by dianakhuang — APPROVED** [review]
Approved from a tech perspective. 

I'm not sure about recommending combining the vertical and discussion blocks into the same file, especially since it will probably export into a separate file in a separate directory, but I don't have strong opinions on it.

Unfortunately, I think this documentation was missed when we replaced the original `discussion` XModule with a new `discussion` XBlock.

**Comment by jolyonb** [comment]
Updated with requested changes. I tested the XML export; discussion blocks are exported with the method indicated as preferred (not in a separate directory). One curious note is that a `url_name` parameter is also exported with the discussion block; this generates warnings in the logs on course import, as the discussion xblock does not have a url_name property.

**Review by grantgoodmanedX — APPROVED** [review]
👍 Looks good.

**Pull request merged**